Dockerfile

```diff
@@ -18,117 +20,129 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
+  && gem install rubocop --version '= 0.39.0' --no-ri --no-rdoc \
+  && gem install bundler --version '= 1.12.1' --no-ri --no-rdoc \
+  && gem install rake --version '= 10.5.0' --no-ri --no-rdoc \
+  && gem install grpc-tools --version '=1.0.0' --no-ri --no-rdoc
 
 # Install couple of git repos
-WORKDIR /
 RUN git clone https://github.com/googleapis/googleapis
 RUN git clone https://github.com/googleapis/toolkit
 ENV TOOLKIT_HOME /toolkit
-WORKDIR /toolkit
-RUN sudo ./gradlew install # Install toolkit. Must sudo to download gradle plugins.
+
+# Install toolkit. Must sudo to download gradle plugins.
+RUN cd /toolkit \
+  && ./gradlew install \
+  && cd /
 
 # Install googleapis protocol compiler plugin which is needed to build gapic resource classes.
-RUN pip install -e git+https://github.com/googleapis/proto-compiler-plugin#egg=remote
-
-# Install PHP protobuf plugin
-WORKDIR /
-RUN apt-get install -y ruby2.0 ruby2.0-dev
-RUN /usr/bin/gem2.0 install ronn --no-ri --no-rdoc
-RUN git clone https://github.com/stanley-cheung/Protobuf-PHP.git
-WORKDIR /Protobuf-PHP
-RUN rake pear:package version=1.0 --trace
-RUN pear install Protobuf-1.0.tgz
-
-# Install PHP formatting tools
-RUN pear install PHP_CodeSniffer-2.7.0
-RUN curl -L https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/v1.11.8/php-cs-fixer.phar -o /usr/local/bin/php-cs-fixer
-RUN chmod a+x /usr/local/bin/php-cs-fixer
-
-# Install artman.
-# TODO(ethanbao): pipeline should be installed from package manager.
-ADD . /src
-WORKDIR /src
-RUN pip install -e .
+RUN pip3 install -e git+https://github.com/googleapis/proto-compiler-plugin#egg=remote
+
+# Install PHP protobuf plugin.
+RUN gem install ronn --no-ri --no-rdoc \
+  && git clone https://github.com/stanley-cheung/Protobuf-PHP.git \
+  && cd /Protobuf-PHP \
+  && rake pear:package version=1.0 --trace \
+  && pear install Protobuf-1.0.tgz \
+
+  # Install PHP formatting tools
+  && pear install PHP_CodeSniffer-2.7.0 \
+  && curl -L https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/v1.11.8/php-cs-fixer.phar -o /usr/local/bin/php-cs-fixer \
+  && chmod a+x /usr/local/bin/php-cs-fixer \
+  && cd /
 
 # Setup git config used by github commit pushing.
 RUN git config --global user.email googleapis-publisher@google.com \
   && git config --global user.name "Googleapis Publisher"
 
 # Setup artman user config
-RUN mkdir /root/.artman
+# Note: This is somewhat brittle as it relies on a specific path
+# outside of or inside Docker.
+#
+# This should probably be fixed to have the smoke test itself provide
+# the configuration.
+# TODO (lukesneeringer): Fix this.
+RUN mkdir -p /root/
 ADD artman-user-config-in-docker.yaml /root/.artman/config.yaml
 
-# Run smoketests
-# TODO(ethanbao): this should be part of artman CI.
-RUN python test/smoketest.py
+# Install artman and run the smoke test.
+RUN pip3 install googleapis-artman==0.4.3
+
+# TODO (lukesneeringer): Move smoke tests to a different venue.
+# RUN smoketest_artman.py \
```

Comment:
I commented this out for now; I am not convinced that the `docker build` is the right place to do this.

You can always run it yourself with:

```
docker run -it googleapis/artman:0.4.3
smoketest_artman.py
```